### PR TITLE
Race condition

### PR DESCRIPTION
**Pull request by acolebourne**

### CODE REVIEW

**Comment by bjagg** [comment]
+1
Is there a jira for this PR?

File: src/main/java/org/jasig/portlet/newsreader/service/SharedNewsSetServiceImpl.java [hunk 1]
```diff
@@ -49,27 +50,33 @@ public NewsSet getNewsSet(String fname, PortletRequest request) {
...
+			// by adding setId to session, we signal that initialisation has taken place.
+	        if (session.getAttribute("setId", PortletSession.PORTLET_SCOPE) == null) {
+	        	log.debug("re-initalising loaded newsSet "+set.getName());
+	            @SuppressWarnings("unchecked")
+	            Set<String> roles = (Set<String>) session.getAttribute("userRoles", PortletSession.PORTLET_SCOPE);
+				
+				newsStore.initNews(set, roles);
+				newsStore.storeNewsSet(set);
+				session.setAttribute("setId", set.getId(), PortletSession.PORTLET_SCOPE);
+			}
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/newsreader/service/SharedNewsSetServiceImpl.java]: Minor, but indent on lines 70 - 77 off.  Should be at same level as 69.

File: src/main/resources/hibernate-mappings/NewsSet.hbm.xml [hunk 2]
```diff
@@ -31,22 +31,24 @@
...
+	        <!-- new set name -->
+	        <property name="name" type="string" not-null="true">
+	            <column name="NEWS_NAME" length="100"/>
+	        </property>
+	
+	        <!-- user id --> 
+	        <property name="userId" type="string" not-null="true">
+	            <column name="USER_ID" length="50" index="user_idx"/>
+	        </property>
+	    </properties>
```
Comment (jameswennmacher) [inline src/main/resources/hibernate-mappings/NewsSet.hbm.xml]: Minor.  indent is off on line 44.

**Comment by jameswennmacher** [comment]
+1.  It would be great to have a Jira and the commits reference the Jira # in the commit message.

**Comment by acolebourne** [comment]
Closed in favour of new PR #39

**Pull request closed without merging**